File: test/com/drawmetry/ecolban/kotlindemo/TestProcess.java
package com.drawmetry.ecolban.kotlindemo;

import org.junit.Test;

import java.util.concurrent.ArrayBlockingQueue;
import java.util.concurrent.BlockingQueue;

import static junit.framework.TestCase.assertEquals;

public class TestProcess {

    @Test
    public void testProcess() throws InterruptedException {
        BlockingQueue<String> inQ = new ArrayBlockingQueue<>(2);
        BlockingQueue<Object> outQ = new ArrayBlockingQueue<>(2);
        inQ.put("Hello world");
        Source<String> source = inQ::take;
        Sink<Object> sink = outQ::put;
        Process sut = new Process(source, sink);
        sut.start();
        String s = (String) outQ.take();
        assertEquals("dlrow olleH", s);
        sut.stop();
    }
}

File: test/com/drawmetry/ecolban/kotlindemo/TestProcess.kt
package com.drawmetry.ecolban.kotlindemo

import org.junit.Test

import java.util.concurrent.ArrayBlockingQueue
import java.util.concurrent.BlockingQueue

import junit.framework.TestCase.assertEquals

class TestProcess {

    @Test
    @Throws(InterruptedException::class)
    fun testProcess() {
        val inQ = ArrayBlockingQueue<String>(2)
        val outQ = ArrayBlockingQueue<Any>(2)
        inQ.put("Hello world")
        val source = Source { inQ.take() }
        val sink = Sink<Any> { outQ.put(it) }
        val sut = Process(source, sink)
        sut.start()
        val s = outQ.take() as String
        assertEquals("dlrow olleH", s)
        sut.stop()
    }
}
